sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py

```diff
@@ -588,122 +588,168 @@ def lift_combiners(stages, context):
 
   ... -> PreCombine -> GBK -> MergeAccumulators -> ExtractOutput -> ...
   """
-  for stage in stages:
-    assert len(stage.transforms) == 1
-    transform = stage.transforms[0]
-    if transform.spec.urn == common_urns.composites.COMBINE_PER_KEY.urn:
-      combine_payload = proto_utils.parse_Bytes(
-          transform.spec.payload, beam_runner_api_pb2.CombinePayload)
+  def is_compatible_with_combiner_lifting(trigger):
```

Comment:
Would you please document the conditions for "being compatible"?

This function is not obvious by reading code: it has several conditions, and is in recursive form.

Reply:
Done.